crates/accelerate/README.md

```diff
@@ -8,3 +8,27 @@ place to put the code.  This is _usually_ the right place to put new Rust/Python
 The `qiskit-pyext` crate is what actually builds the C extension modules.  Modules in here should define
 themselves has being submodules of `qiskit._accelerate`, and then the `qiskit-pyext` crate should bind them
 into its `fn _accelerate` when it's making the C extension.
+### Adding a module to `_accelerate`
+Specifically, adding a rust module `my_module` to Qiskit such that it can be called from within Python involves
+the following steps:
+1. Add a module function to your to your `my_module.rs` file, it should look similar to this:
+  
+     ```rust
+     pub fn my_module(m: &Bound<PyModule>) -> PyResult<()> {}
+     ```
+     The body of this function should add each class/method/function that you want to be exposed to python through this module. It should look similar to this:
+     ```rust
+     pub fn my_module(m: &Bound<PyModule>) -> PyResult<()> {
+        m.add_wrapped(wrap_pyfunction!(mymodulefunction))?; // For functions.
+        m.add_class::<MyModuleClass>()?; // For classes.
+     }
+     ```
+2. To expose the module add `pub mod my_module` at the header of `crates/accelerate/src/lib.rs`.
+   ```rust
+   pub mod mymodule;
+   ...
+3. To `crates/pyext/src/lib.rs` 
+   * Add my_module::my_module to `use qiskit_accelerate::{`
+   * Add `add_submodule(m, my_module, "my_module")?;`
```

Comment:
The second bullet doesn't explain where that line goes, though hopefully people would be able to tell.

As of #12889, the second bullet is also a little bit out-of-date - could you update it for now, and when we upgrade to PyO3 0.22, we'll change it to the preferred form for that.